src/test/java/org/takes/tk/TkCORSTest.java

```diff
@@ -0,0 +1,127 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.takes.tk;
+
+import java.util.HashSet;
+import java.util.Set;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.takes.Request;
+import org.takes.Take;
+import org.takes.rq.RqFake;
+import org.takes.rq.RqWithHeaders;
+import org.takes.rs.RsPrint;
+import org.takes.rs.RsText;
+
+/**
+ * Test case for {@link TkCORS}.
+ * @author Endrigo Antonini (teamed@endrigo.com.br)
+ * @version $Id$
+ * @since 0.20
+ */
+public final class TkCORSTest {
+
+    /**
+     * TkCORS can handle a connection without origin in the request.
+     * @throws Exception If some problem inside
+     */
+    @Test
+    public void withoutOrigin() throws Exception {
+        final Take take = Mockito.mock(Take.class);
+        Mockito.doReturn(new RsText()).when(take)
+            .act(Mockito.any(Request.class));
+        final Take cors = new TkCORS(take);
+        final RsPrint response = new RsPrint(cors.act(new RqFake()));
+        final String head = response.printHead();
+        this.httpStatusOk(head);
+    }
+
+    /**
+     * TkCORS can handle connections with correct domain on origin.
+     * @throws Exception If some problem inside
+     */
+    @Test
+    public void withCorrectDomainOnOrigin() throws Exception {
+        final Take take = Mockito.mock(Take.class);
+        Mockito.doReturn(new RsText()).when(take)
+            .act(Mockito.any(Request.class));
+        final Take cors = new TkCORS(
+            take,
+            "http://teamed.io",
+            "http://example.com"
+        );
+        final Set<String> headers = new HashSet<String>();
+        headers.add("Origin: http://teamed.io");
+        final Request req = new RqWithHeaders(new RqFake(), headers);
+        final RsPrint response = new RsPrint(cors.act(req));
+        final String head = response.printHead();
+        this.httpStatusOk(head);
+    }
+
+    /**
+     * TkCors can't handle connections with wrong domain on origin.
+     * @throws Exception If some problem inside
+     */
+    @Test
+    public void withWrongDomainOnOrigin() throws Exception {
+        final Take take = Mockito.mock(Take.class);
```

Comment:
@antonini mocking is a bad practice in general, try to use real objects, see http://www.yegor256.com/2014/09/23/built-in-fake-objects.html